[2.2.0] test_writemap_newdefault fails on x86_64 with Astropy 8.0.0b1

I made a test build of [Astropy 8.0.0b1](https://github.com/astropy/astropy/releases/tag/v8.0.0b1) on Debian experimental, which included a run of the automated tests for all reverse dependencies. For healpy, this resulted in one failiure of x86_64:

```
___________________________ test_writemap_newdefault ___________________________

tmp_path = PosixPath('/tmp/pytest-of-debci/pytest-0/test_writemap_newdefault0')

    def test_writemap_newdefault(tmp_path):
        filename = tmp_path / "test_map.fits"
        for dtype in (
            "bool",
            "uint8",
            "int16",
            "int32",
            "int64",
            "float32",
            "float64",
            "complex64",
            "complex128",
        ):
            m = np.ones(12, dtype=dtype)
            write_map(filename, m, overwrite=True)
            with pf.open(filename) as f:
                assert f[1].data.field(0).dtype.name == dtype
>               np.testing.assert_array_almost_equal(f[1].data.field(0), m)
E               AssertionError: 
E               Arrays are not almost equal to 6 decimals
E               
E               Mismatched elements: 12 / 12 (100%)
E               First 5 mismatches are at indices:
E                [0]: 1j (ACTUAL), (1+0j) (DESIRED)
E                [1]: 1j (ACTUAL), (1+0j) (DESIRED)
E                [2]: 1j (ACTUAL), (1+0j) (DESIRED)
E                [3]: 1j (ACTUAL), (1+0j) (DESIRED)
E                [4]: 1j (ACTUAL), (1+0j) (DESIRED)
E               Max absolute difference among violations: 1.4142135381698608
E               Max relative difference among violations: 1.4142135381698608
E                ACTUAL: array([ 0.+1.j,  0.+1.j,  0.+1.j,  0.+1.j,  0.+1.j,  0.+1.j,  0.+1.j,
E                       0.+1.j,  0.+1.j,  0.+1.j,  0.+1.j,  0.+1.j], dtype='>c8')
E                DESIRED: array([ 1.+0.j,  1.+0.j,  1.+0.j,  1.+0.j,  1.+0.j,  1.+0.j,  1.+0.j,
E                       1.+0.j,  1.+0.j,  1.+0.j,  1.+0.j,  1.+0.j], dtype=complex64)

test/test_fitsfunc.py:393: AssertionError
```
Full test log [here](https://ci.debian.net/packages/h/healpy/unstable/amd64/71378484/).

There is another failure on arm64, for which I creates #1102. Both are regressions, i.e. they do not happen with Astropy 7.2.

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -37,7 +37,7 @@ jobs:
       - uses: astral-sh/setup-uv@v7
 
       - name: Setup ccache
-        uses: hendrikmuhs/ccache-action@v1.2
+        uses: hendrikmuhs/ccache-action@v1.2.18
         with:
           key: ${{ runner.os }}-py${{ matrix.python-version }}
           max-size: 256M

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -36,7 +36,16 @@ jobs:
 
       - uses: astral-sh/setup-uv@v7
 
+      - name: Setup ccache
+        uses: hendrikmuhs/ccache-action@v1.2
+        with:
+          key: ${{ runner.os }}-py${{ matrix.python-version }}
+          max-size: 256M
+
       - name: Install dependencies
+        env:
+          CC: "ccache gcc"
+          CXX: "ccache g++"
         run: |
           uv venv .venv
           if [ "${{ matrix.astropy-prerelease }}" = "true" ]; then

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -1,0 +1,55 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+  schedule:
+    # Weekly on Tuesday at 9 AM Pacific (16:00 UTC)
+    - cron: '0 16 * * 2'
+
+jobs:
+  test:
+    name: Python ${{ matrix.python-version }}${{ matrix.astropy-prerelease && ' (astropy pre-release)' || '' }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.10", "3.11", "3.12", "3.13"]
+        astropy-prerelease: [false]
+        include:
+          - python-version: "3.13"
+            astropy-prerelease: true
+
+    steps:
+      - uses: actions/checkout@v6
+        with:
+          fetch-depth: 0
+          submodules: recursive
+
+      - uses: actions/setup-python@v6
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - uses: astral-sh/setup-uv@v7
+
+      - name: Install dependencies
+        run: |
+          uv venv .venv
+          if [ "${{ matrix.astropy-prerelease }}" = "true" ]; then
+            uv pip install --python .venv/bin/python --pre ".[test]" "astropy>=8.0.0.dev"
+          else
+            uv pip install --python .venv/bin/python ".[test]"
+          fi
+
+      - name: Run doctests
+        run: |
+          source .venv/bin/activate
+          pytest --doctest-plus --doctest-cython --pyargs healpy
+
+      - name: Run test suite
+        run: |
+          source .venv/bin/activate
+          pytest test/

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -14,6 +14,7 @@ jobs:
   test:
     name: Python ${{ matrix.python-version }}${{ matrix.astropy-prerelease && ' (astropy pre-release)' || '' }}
     runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.astropy-prerelease }}
     strategy:
       fail-fast: false
       matrix:

--- a/AGENTS.md
+++ b/AGENTS.md
@@ -27,5 +27,10 @@ Commits follow a short, imperative summary (`Fix query_disc strip for NEST`). Gr
 
 **Never push directly to `main` without explicit user approval.** Always show the proposed changes and ask for confirmation before pushing to any protected branch.
 
+## CI Workflows
+- **Tests** (`.github/workflows/tests.yml`): Runs on every push to `main` and on pull requests. Matrix: Python 3.10–3.13 with stable astropy, plus one job with astropy pre-release (`continue-on-error: true`). Uses ccache to speed up C++ recompilation on repeat runs. Total job time ~4–5m cold, ~2m with ccache hits.
+- **Build & Publish** (`.github/workflows/cibuildwheel.yml`): Builds source dist and binary wheels for multiple platforms. Runs cibuildwheel smoke tests inside each wheel. Publishes to PyPI on GitHub release.
+- **Editable Install Check** (`.github/workflows/uv-editable-install.yml`): Verifies `uv pip install -e .` succeeds on PRs.
+
 ## Release Process
 Follow the checklist in `RELEASE.md` for tagging, wheel builds, and PyPI uploads. Confirm CI artifacts match the documented version bump and update communication channels as described there.

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -1,5 +1,6 @@
 Unreleased
 
+* CI: Added test workflow (`.github/workflows/tests.yml`) running the full test suite on Python 3.10–3.13 with ccache, plus an astropy pre-release job https://github.com/healpy/healpy/pull/1109
 * Fix off-by-one in ``synalm`` Cℓ boundary check that caused out-of-bounds reads when ``l == len(cl)`` https://github.com/healpy/healpy/pull/1108
 
 Release 1.20.0b1 29 May 2026

--- a/README.rst
+++ b/README.rst
@@ -11,6 +11,9 @@ Healpy, a python wrapper for healpix
 .. image:: https://github.com/healpy/healpy/actions/workflows/cibuildwheel.yml/badge.svg
    :target: https://github.com/healpy/healpy/actions/workflows/cibuildwheel.yml
 
+.. image:: https://github.com/healpy/healpy/actions/workflows/tests.yml/badge.svg
+   :target: https://github.com/healpy/healpy/actions/workflows/tests.yml
+
 .. image:: https://readthedocs.org/projects/healpy/badge/?version=latest
    :target: https://readthedocs.org/projects/healpy/?badge=latest
    :alt: Documentation Status

